modules/orionode.debug.server/lib/protocol.js

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
```

Comment:
This is the wire protocol behind the debugger. It's very similar to the VS Code language server one.
VS Code provides a node package with this wire protocol implementation (`vscode-debugadapter/lib/protocol` see below) to help debugger adapter developers. We use this here to implement the debugger host (the role of VS Code).
It contains three types of message: requests, responses and events. Requests and responses can be sent in both directions, however events can be only sent from adapters to the debugger host. As a result, this wire protocol implementation (`vscode-debugadapter/lib/protocol`) doesn't support receiving events, because an adapter doesn't need to.
Therefore we need to modify the source code of this implementation to support receiving events. The change is very small (line 52 - 54) but is nested deeply.
CQ has been approved.